src/data/bcc_datamodule.py

```diff
@@ -0,0 +1,177 @@
+        batch_size (int): The batch size for data loaders. Defaults to 64.
+        num_workers (int): The number of workers for data loading. Defaults to 0.
+        pin_memory (bool): Whether to pin memory during data loading. Defaults to False.
+
+    Methods:
+        num_classes(): Get the number of classes, which is 3 for Breast Cancer classification.
+        setup(stage: Optional[str] = None): Load the data for the specified stage (train, validation, test, or predict).
+        train_dataloader(): Get a DataLoader for the training data.
+        val_dataloader(): Get a DataLoader for the validation data.
+        test_dataloader(): Get a DataLoader for the test data.
+
+    Note:
+        This data module assumes that the dataset loader 'DeepLakeDataset' is available, and that it is compatible with both local directory paths and deeplake datasets.
+    """
+
+    def __init__(
+        self,
+        train_dir: Union[str, deeplake.dataset],  # allow for both str and deeplake.dataset
+        val_dir: Union[str, deeplake.dataset],
+        test_dir: Union[str, deeplake.dataset],
+        input_size: Tuple[int, int] = [600, 500],
+        batch_size: int = 64,
+        num_workers: int = 0,
+        pin_memory: bool = False,
+    ):
+        super().__init__()
+
+        self.save_hyperparameters(logger=False)
+
+        self.train_transform = transforms.Compose([
+            transforms.ToPILImage(),
+            transforms.Grayscale(num_output_channels=1),
+            transforms.Resize((256, 256)),
+            transforms.RandomRotation(20),
+            transforms.ToTensor(),
+            transforms.Normalize(mean=[0], std=[1])
+        ])
+
+        self.val_transform = transforms.Compose([
+            transforms.ToPILImage(),
+            transforms.Grayscale(num_output_channels=1),
+            transforms.Resize((256, 256)),
+            transforms.ToTensor(),
+            transforms.Normalize(mean=[0], std=[1])
+        ])
+
+        self.test_transform = transforms.Compose([
+            transforms.ToPILImage(),
+            transforms.Grayscale(num_output_channels=1),
+            transforms.Resize((256, 256)),
+            transforms.ToTensor(),
+            transforms.Normalize(mean=[0], std=[1])
+        ])
+
+        self.ds_train = train_dir
+        self.ds_val = val_dir
+        self.ds_test = test_dir
+
+        # init storage to save data
+
+        self.data_train: Optional[Dataset] = None
+        self.data_val: Optional[Dataset] = None
+        self.data_test: Optional[Dataset] = None
+
+    @property
+    def num_clasess(self):
+        """Get the number of classes, which is 3 for Breast Cancer classification.
+
+        Returns:
+            int: The number of classes, which is 3 for Breast Cancer classification.
+        """
+        return 3
+
+    def setup(self, stage: Optional[str] = None):
+        """Load the data for the specified stage (train, validation, test, or predict).
+
+        Args:
```

Comment:
not necessary, incorrect and inconsistent docstring style, please use numpydocstyle

Reply:
please review it again